reps/2023-10-13-accelerator-support.md

```diff
@@ -0,0 +1,234 @@
+def task()
+    ...
+
+@ray.remote(num_accelerators=1, accelerator_family="TPU")
+def task()
+    ...
+
+@ray.remote(num_accelerators=1, accelerator_type="TPU-V4")
+def task()
+    ...
+```
+
+#### Implementation
+
+Core defines an `Accelerator` base class and each accelerator needs to implement a subclass (e.g. `TPUAccelerator`):
+
+```python
+@DeveloperAPI
+class Accelerator(ABC):
+    @staticmethod
+    @abstractmethod
+    def is_available() -> bool:
+        """Detect if the accelerator is available on this node."""
+
+    @staticmethod
+    @abstractmethod
+    def get_resource_name() -> str:
+        """Get the name of the resource representing this accelerator."""
+
+    @staticmethod
+    @abstractmethod
+    def get_visible_accelerator_ids_env_var() -> str:
+        """Get the env var that sets the ids of visible accelerators."""
+
+    @staticmethod
+    @abstractmethod
+    def get_num_accelerators() -> int:
+        """Get the total number of accelerators on this node."""
+
+    @staticmethod
+    @abstractmethod
+    def get_visible_accelerator_ids() -> Optional[List[str]]:
+        """Get the ids of accelerators visible to Ray."""
+
+    @staticmethod
+    @abstractmethod
+    def get_accelerator_type() -> Optional[str]:
+        """Get the type of the accelerator on this node.
+
+        The result should only be used when get_num_accelerators() > 0.
+
+        Return None if it's unknown.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def validate_resource_request_quantity(quantity: float) -> Optional[str]:
+        """Validate the resource request quantity of this accelerator resource.
+
+        Return error message if the quantity is invalid.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def set_visible_accelerator_ids(ids: List[str]) -> None:
+        """Set the ids of accelerators visible to the Ray task or actor."""
+
+    @staticmethod
+    def get_ec2_num_accelerators(instance_type: str, instances: dict) -> Optional[int]:
+        """Get the number of accelerators of this aws ec2 instance type.
+
+        Return None if it's unknown.
+        """
+        return None
+
+    @staticmethod
+    def get_ec2_accelerator_type(instance_type: str, instances: dict) -> Optional[str]:
+        """Get the accelerator type of this aws ec2 instance type.
+
+        Return None if it's unknown.
+        """
+        return None
+```
+
+Besides defining the accelerator subclass, the accelerator resource needs to be marked as unit instance resource in config `RAY_custom_unit_instance_resources`.
+
+### Ray Train
+TODO
+
+### Ray RLLib
+TODO
+
+### Ray Serve
+Nothing needs to be changed on the serve side.
+
+Users can define a deployment using accelerators via `ray_actor_options`:
+
+```python
+@serve.deployment(ray_actor_options={"num_gpus": 1})
+class Deployment:
+    ...
+
+@serve.deployment(ray_actor_options={"resources": {"TPU": 1}})
+class Deployment:
+    ...
+```
+
+### Ray Data
```

Comment:
@raulchen for reviewing the data section.

Reply:
> `ds.map_batches(Predicator, resources={"TPU": 1})`

The correct syntax should be `ds.map_batches(..., ray_remote_args={"resources": {"TPU": 1}})`
Also, we've deprecated the `Predicator` API. So better not to add it in the example. 
Otherwise looks good.